src/python/pants/core/goals/update_build_files_test.py

```diff
@@ -184,6 +187,58 @@ def test_black_config(black_rule_runner: RuleRunner) -> None:
     assert Path(black_rule_runner.build_root, "BUILD").read_text() == "tgt(name='t')\n"
 
 
+# ------------------------------------------------------------------------------------------
+# Yapf formatter fixer
+# ------------------------------------------------------------------------------------------
+
+
+@pytest.fixture
+def yapf_rule_runner() -> RuleRunner:
+    return RuleRunner(
+        rules=(
+            format_build_file_with_yapf,
+            update_build_files,
+            *pants_bin.rules(),
+            *config_files.rules(),
+            *pex.rules(),
+            SubsystemRule(Yapf),
+            SubsystemRule(UpdateBuildFilesSubsystem),
+            UnionRule(RewrittenBuildFileRequest, FormatWithYapfRequest),
+        )
+    )
+
+
+def test_yapf_fixer_fixes(yapf_rule_runner: RuleRunner) -> None:
+    yapf_rule_runner.write_files({"BUILD": "tgt( name =  't' )"})
+    result = yapf_rule_runner.run_goal_rule(UpdateBuildFilesGoal, env_inherit=BLACK_ENV_INHERIT)
```

Comment:
I spent hours trying to find how to pass the `--in-place` argument to this, but failed :( I am sure you have some bright ideas? :)